OpenKH/egsTools.cs

```diff
@@ -463,6 +501,119 @@ private static List<EgsHdAsset.RemasteredEntry> ReplaceRemasteredAssets(string i
+
+                    assetData = File.ReadAllBytes(assetFilePath);
+                    decompressedLength = assetData.Length;
+                    assetData = remasteredAssetHeader.CompressedLength > -1 ? Helpers.CompressData(assetData) : assetData;
+                    assetData = remasteredAssetHeader.CompressedLength > -2 ? EgsEncryption.Encrypt(assetData, seed) : assetData;
+                    if (imd != null && !imd.Invalid) originalAssetOffset = imd.Offset;
+                }
+                else
+                {
+                    Console.WriteLine($"Keeping remastered file: {relativePath}/{filename}");
+                    // The original file have been replaced, we need to encrypt all remastered asset with the new key
+                    if (!seed.SequenceEqual(asset.Seed))
+                    {
+                        assetData = remasteredAssetHeader.CompressedLength > -1 ? Helpers.CompressData(assetData) : assetData;
+                        assetData = remasteredAssetHeader.CompressedLength > -2 ? EgsEncryption.Encrypt(assetData, seed) : assetData;
+                        if (imd != null && !imd.Invalid && imd.TextureCount >= i) originalAssetOffset = imd.Offset;
+                    }
+                    else
+                    {
+                        assetData = asset.RemasteredAssetsCompressedData.ContainsKey(filename) ? asset.RemasteredAssetsCompressedData[filename] : new byte[] { };
+                    }
+                }
+                var compressedLength = remasteredAssetHeader.CompressedLength > -1 ? assetData.Length : remasteredAssetHeader.CompressedLength;
+
+                var newRemasteredAssetHeader = new EgsHdAsset.RemasteredEntry()
+                {
+                    CompressedLength = compressedLength,
+                    DecompressedLength = decompressedLength,
+                    Name = filename,
+                    Offset = offset,
+                    OriginalAssetOffset = originalAssetOffset
+                };
+
+                newRemasteredHeaders.Add(newRemasteredAssetHeader);
+
+                // Write asset header in the PKG stream
+                BinaryMapping.WriteObject<EgsHdAsset.RemasteredEntry>(pkgStream, newRemasteredAssetHeader);
+
+                // Don't write into the PKG stream yet as we need to write
+                // all HD assets header juste after original file's data
+                allRemasteredAssetsData.Write(assetData);
+
+                // Make sure to align remastered asset data on 16 bytes
+                if (assetData.Length % 0x10 != 0)
+                {
+                    allRemasteredAssetsData.Write(Enumerable.Repeat((byte)0xCD, 16 - (assetData.Length % 0x10)).ToArray());
+                }
+
+                offset += decompressedLength;
+            }
+
+            pkgStream.Write(originalAssetData);
+            pkgStream.Write(allRemasteredAssetsData.ReadAllBytes());
+
+            return newRemasteredHeaders;
+        }
+
```

Comment:
if we handle a SDasset rather than MDLX/IMD, you can keep using the same ReplaceRemasteredAssets function (rather than duplicating 99% of its content) here.